packages/common/src/tools/queue/Retries.ts

```diff
@@ -0,0 +1,47 @@
+import { ShouldRetry } from './types'
+
+function always() {
+  return { retry: true }
+}
+
+interface ExponentialBackOffOpts {
+  maxAttempts?: number
+  maxDistance?: number
+  now?: typeof Date['now']
+}
+
+export function exponentialBackOff(
+  stepMs: number,
+  opts?: ExponentialBackOffOpts,
+) {
+  const maxAttempts = opts?.maxAttempts ?? Infinity
+  const maxDistance = opts?.maxDistance ?? Infinity
+  const now = opts?.now ?? Date.now
+
+  return ({ attempts }: { attempts: number }) => {
+    if (attempts === maxAttempts) {
+      return {
+        retry: false,
+      }
+    }
+    const distance = Math.pow(2, attempts) * stepMs
+    return {
+      retry: true,
+      executeAfter: now() + Math.min(distance, maxDistance),
```

Comment:
Maybe the code would be simpler to test if we returned only the `Math.min(distance, maxDistance)` value and added `now()` inside the TaskQueue? Then exponentialBackOff becomes pure.

Reply:
I like it, will do that 👍